clients/venice-thin-client/src/main/java/com/linkedin/venice/client/schema/RouterBasedStoreSchemaFetcher.java

```diff
@@ -57,15 +58,40 @@ public Schema getKeySchema() {
 
   @Override
   public Schema getLatestValueSchema() {
+    // Fetch the latest value schema for the specified value schema.
+    String latestSchemaRequestPath = TYPE_LATEST_VALUE_SCHEMA + "/" + storeClient.getStoreName();
+    try {
+      String latestSchemaStr = fetchSingleSchemaString(latestSchemaRequestPath);
+      Schema latestSchema;
+      try {
+        latestSchema = parseSchemaFromJSONLooseValidation(latestSchemaStr);
+      } catch (Exception e) {
+        throw new VeniceException("Got exception while parsing superset schema", e);
+      }
+      return latestSchema;
+    } catch (Exception e) {
```

Comment:
These two layers of `try/catch` is a little confusing.
What is your intention here? If we want to fall back to `getLatestValueSchemaFromAllValueSchemas` if any issue happens to the logic of the body of `try/catch`, one layer of `try/catch` is good enough.
If the intention is to skip the fallback if there is a schema parsing error, we should throw an specific exception and pop it up in the outer layer of `try/catch`, right?

Reply:
I can (and should) wrap just the `fetchSingleSchemaString` call in the first try catch

Reply:
The intention is:
1. Try the new API: `/latest_schema/storeName`
    1. If it is available on routers, it will return a schema. Use that
    2. If it is not available on routers, the API will return a 400 response. This is thrown as an exception in executeRequest. If this happens, call the `/value_schema/storeName` API and infer the superset schema from that
    3. The two APIs calls are only made to support backward compatibility. Once sufficient time has passed, we can remove the call to `/value_schema` endpoint